src/main/java/uk/ac/ebi/ampt2d/persistence/entities/FileMetadata.java

```diff
@@ -0,0 +1,157 @@
+
+import javax.persistence.CascadeType;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.EntityListeners;
+import javax.persistence.EnumType;
+import javax.persistence.Enumerated;
+import javax.persistence.FetchType;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+import javax.persistence.OneToMany;
+import javax.persistence.Table;
+import javax.persistence.Transient;
+import javax.validation.constraints.Size;
+import java.io.File;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.util.Set;
+
+@Entity
+@EntityListeners(AuditingEntityListener.class)
+@Table(name="file")
+public class FileMetadata {
+    public static final int MIN_FILE_HASH = 1;
+
+    public static final int MAX_FILE_HASH = 128;
+
+    @Id
+    @GeneratedValue
+    private long id;
+
+    @Column(length = MAX_FILE_HASH, unique = true)
+    @Size(min = MIN_FILE_HASH, max = MAX_FILE_HASH)
+    private String hash;
+
+    @Enumerated(EnumType.STRING)
+    private Type type;
+
+    private String name;
+
+    private long size;
+
+    @OneToMany(cascade = CascadeType.ALL, mappedBy = "fileMetadata", fetch = FetchType.EAGER)
+    private Set<SourceFilePath> sourceFilePaths;
+
+    @CreatedDate
+    @Column(nullable = false, updatable = false)
+    private LocalDateTime createdDate;
+
+    @LastModifiedDate
+    @Column(nullable = false)
+    private LocalDateTime lastModifiedDate;
+
+    public FileMetadata(String hash, String name, Type type, long size) {
+        this.hash = hash;
+        this.name = name;
+        this.type = type;
+        this.size = size;
+    }
+
+    public FileMetadata(File file, Type type, String name) throws IOException {
+        this.hash = Files.hash(file, Hashing.sha384()).toString();
+        this.type = type;
+        this.size = file.length();
+        this.name = name;
+    }
+
+    public FileMetadata() {
+    }
+
+    public String getHash() {
+        return hash;
+    }
+
+    public void setHash(String hash) {
+        Assert.hasText(hash, "Hash is required");
+        this.hash = hash;
+    }
+
+    public Type getType() {
+        return type;
+    }
+
+    public void setType(Type type) {
+        this.type = type;
+    }
+
+    public long getSize() {
+        return size;
+    }
+
+    public void setSize(long size) {
+        this.size = size;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        Assert.hasText(name, "File name is required");
+        this.name = name;
+    }
+
+    public Set<SourceFilePath> getSourceFilePaths() {
+        return sourceFilePaths;
+    }
+
+    public void setSourceFilePaths(Set<SourceFilePath> sourceFilePaths) {
+        if (sourceFilePaths != null) {
+            for (SourceFilePath sourceFilePath : sourceFilePaths) {
+                sourceFilePath.setFileMetadata(this);
+            }
+            this.sourceFilePaths = sourceFilePaths;
```

Comment:
Don't store directly the collection, add all the items of the collection to your own. Initializate the sourceFilePaths in constructor.